UnityProject/Assets/Scripts/Systems/Explosions/Explosion.cs

```diff
@@ -85,28 +87,29 @@ public void Step()
 				return;
 			}
 
-			var WorldPSos = new Vector3Int(x0, y0, 0);
-			var Matrix = MatrixManager.AtPoint(WorldPSos, CustomNetworkManager.IsServer);
-			var Local = WorldPSos.ToLocal(Matrix.Matrix).RoundToInt();
-			var NodePoint = Matrix.MetaDataLayer.Get(Local); //Explosion node
+			Vector3Int WorldPSos = new Vector3Int(x0, y0, 0);
+			MatrixInfo Matrix = MatrixManager.AtPoint(WorldPSos, CustomNetworkManager.IsServer);
+			Vector3Int Local = WorldPSos.ToLocal(Matrix.Matrix).RoundToInt();
+			MetaDataNode NodePoint = Matrix.MetaDataLayer.Get(Local); //Explosion node
+
 			if (NodePoint != null)
 			{
-				if (NodePoint.ExplosionNode == null)
+				if (!NodePoint.ExplosionNodes.Any(p => p.GetType() == NodeType.GetType()))
 				{
-					NodePoint.ExplosionNode = new ExplosionNode();
-					NodePoint.ExplosionNode.Initialise(Local, Matrix.Matrix);
```

Comment:
Should reuse the explosion nodes on the tile

Reply:
wdym?